slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/jwt/JwtFilter.java

```diff
@@ -0,0 +1,67 @@
+package com.binarystudio.academy.slidez.infrastructure.security.jwt;
+
+import com.binarystudio.academy.slidez.domain.user.UserService;
+import com.binarystudio.academy.slidez.domain.user.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Optional;
+
+import static org.springframework.util.StringUtils.hasText;
+
+@Component
+public class JwtFilter extends OncePerRequestFilter {
+
+	private final String[] PUBLIC_URLS = { "/", "/health", "/auth/**", "/swagger-ui/**", "/api-docs/**" };
+
+	private final JwtProvider jwtProvider;
+
+	private final UserService userService;
+
+	@Autowired
+	public JwtFilter(JwtProvider jwtProvider, UserService userService) {
+		this.jwtProvider = jwtProvider;
+		this.userService = userService;
+	}
+
+	@Override
+	protected boolean shouldNotFilter(HttpServletRequest request) {
+		for (String excludedPath : PUBLIC_URLS) {
+			if (request.getRequestURI().startsWith(excludedPath)) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	@Override
+	protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+			throws ServletException, IOException {
+		String token = getTokenFromRequest(request);
+		if (token != null) {
+			Optional<User> user = userService.getByToken(token);
```

Comment:
One of the benefits of JWT is that it allows you to not fetch user every request, which we essentially throw out of the window